### DISCUSSION TITLE
How to Know Left Available Capacity?

### DISCUSSION THREAD

**Discussion by LawyZheng** [minio/minio#15889] [Q&A] [answered]
Thanks to MinIO I can easily build an object storage. 

But I notice that there is no way to know about the left available capacity, while only the used capacity is shown in the MinIO Console. I know MinIO is able to set the quota of each bucket, but it doesn't meet my requirement.

I can get the left available capacity from the system disk when I use the SNSD mode, but I don't know how to get this information when I use SNMD or MNMD mode. As far as I know, the Erasure Code will spend extra space, so the available capacity can't be the simple sum of the left capacity of every drive.

I find the calculator provided by MinIO Homepage. Unfortunately, it can only be used in the planning stage though I haven't understood how to use it clearly, for example, it can't calculate the 3 nodes with 6 drives(though the recommended minimum is 4 nodes). But I want to how I can get the left available capacity in real-time. 

Can I get the left available capacity in the MinIO Console, or use `mc` client?
Or can you provide the formula for the available capacity calculation, so I can calculate by myself?

Thanks anyway.

**Comment by mshanmu** [comment, marked as answer]
You can find the information by using mc admin info --json <alias> , search for "totalspace", "usedspace", "availspace" fields on a per drive basis.

Also, you can find the information in prometheus metrics, for example, for usable capacity - minio_cluster_capacity_usable_free_bytes variable.

**Reply by LawyZheng** [reply]
Thanks for the reply.

I do find the information in the `mc admin info -json`, but it seems to be the physical disk space(not excluding the Erasure Code effect). So it doesn't meet my requirements.

I also tried the Prometheus Metrics.  I tried some tests, but unfortunately, I still can't understand how it works.
Here is my case:
I got 3 servers with 50G(excluding the usage of the OS, maybe decreased to 44G) disk space in each one, so I got about 120G in total.
I run an MSMD MinIO cluster with 3 nodes and 6 drives (each node with 2 drivers).
However, when I query the Prometheus Metrics, I got some metrics as follows:
`minio_cluster_capacity_raw_free_bytes` shows about 31.17G in each server.
`minio_cluster_capacity_raw_total_bytes` shows about 33.82G in each server.
`minio_cluster_capacity_usable_total_bytes` shows about 16.91G in each server.
As for `minio_cluster_capacity_usable_free_bytes`, I tried to upload a 30G file, expecting to see the total usable free space of 3 servers could decrease by 30G. Unluckily, it didn't meet my expectation, only decreasing by about 22G actually.

So I'm a little confused about this situation.
Which information could be trusted?

**Reply by vadmeste** [reply]
@LawyZheng can you show the value of both of those:

```
minio_cluster_capacity_usable_total_bytes
minio_cluster_capacity_usable_free_bytes
```

also what is your MinIO version ?

**Reply by LawyZheng** [reply]
@vadmeste 
My MinIO version is `RELEASE.2022-10-15T19-57-03Z (go1.18.7 linux/amd64)`

When I set all nodes in default EC,
`minio_cluster_capacity_usable_total_bytes` comes to `145261670400` in each node metrics.
`minio_cluster_capacity_usable_free_bytes` comes to `138179723264` in each node metrics.

When I set all nodes in EC:2,
`minio_cluster_capacity_usable_total_bytes` comes to `193682227200` in each node metrics.
`minio_cluster_capacity_usable_free_bytes` comes to `184241606656` in each node metrics.

**Reply by echo-devnull** [reply]
@LawyZheng Have you been able to figure this out?

It's... "interesting"  that this is not easily viewable with either the web console or the cli?

**Reply by LawyZheng** [reply]
> @LawyZheng Have you been able to figure this out?
> 
> It's... "interesting" that this is not easily viewable with either the web console or the cli?

I still don't know the best solution to my question, since this question is still marked as "unanswered". 

Though I could get the date from the prometheus metrics, I couldn't get the correct number as I expected.
I don't know whether MinIO has fixed it in the latest version or not.

**Comment by rnic92** [comment]
If you're in the UI and you go to monitoring -> metrics -> info (tab at the top) you can see the drives available space/ used space/ and some more information

**Comment by sadum-vunet** [comment]
Can we get this information to python?
